python/ray/serve/_private/replica.py

```diff
@@ -1204,6 +1218,52 @@ def get_num_ongoing_requests(self) -> int:
         """
         return self._replica_impl.get_num_ongoing_requests()
 
+    def list_outbound_deployments(self) -> List[DeploymentID]:
+        """List all outbound deployment IDs this replica calls into."""
+        seen_deployment_ids: Set[DeploymentID] = set()
+
+        # First, collect dynamically created handles
+        for deployment_id in self._replica_impl.get_dynamically_created_handles():
+            seen_deployment_ids.add(deployment_id)
+
+        # Get the initialized user callable
+        user_callable = self._replica_impl._user_callable_wrapper._callable
+        if user_callable is None:
+            # Callable hasn't been initialized yet, return only dynamic handles
+            return list(seen_deployment_ids)
+
+        # Inspect all attributes of the user callable for stored handles
+        def _find_handles_in_object(obj: Any, visited: set) -> None:
```

Comment:
why not grab handles from the init args? when we compose deployments the handles are injected as class init arguments